src/Pact/Analyze/Check.hs

```diff
@@ -63,21 +60,25 @@ import           Pact.Analyze.Term
 import           Pact.Analyze.Translate
 import           Pact.Analyze.Types
 
-data CheckFailure
-  = Invalid SBVI.SMTModel
-  | Unsatisfiable
-  | Unknown String -- reason
-  | SatExtensionField SBVI.SMTModel
-  | ProofError [String]
-  | TypecheckFailure (Set TC.Failure)
-  | AnalyzeFailure AnalyzeFailure
-  | TranslateFailure TranslateFailure
-  | PropertyParseError Exp
-  --
-  -- TODO: maybe remove this constructor from from CheckFailure.
-  --
-  | CodeCompilationFailed String
-  deriving (Show)
+data CheckSuccess where
+  SatisfiedProperty :: SBVI.SMTModel -> CheckSuccess
+  ProvedTheorem     ::                  CheckSuccess
+
```

Comment:
I switched these to GADT syntax because for a little while during my refactor I was indexing `Check`, `CheckSuccess`, and `CheckFailure` by the new type `Goal`.

Reply:
Decided not to do this, because then we'd have to deal with existential wrappers for heterogenous lists of `Check`s.

Reply:
Do you prefer GADT syntax here?

Reply:
No preference here. you?

Reply:
Slight preference for non-GADT

Reply:
I'll change them back

Reply:
Done